tour/exercise-1.md

```diff
@@ -297,113 +295,357 @@ observer as follows.
+    wordCounter = new WordCounter(context.getAppConfiguration());
+  }
+
+  private void adjustCounts(TransactionBase tx, int delta, String[] words) {
+    wordCounter.adjustCounts(tx, delta, words);
+  }
+```
+
+The CFM groups key values into buckets for efficiency and processes the updates for entire bucket in
+a single transaction.  When you run this new code, that is why `== begin CFM updates ==` is seen at
+least three times for each group of documents loaded.
+
+When you run this example you will also notice two new prefixes in the output of the table scan.  First
+the `wc:` prefix is where the CFM stores its data.  By default the CFM uses Kryo for serialization
+and therefore the key values with this prefix contain non-ASCII characters.  The utility function
+`FluoITHelper.printFluoTable()` escapes non-ASCII characters with `\x<HEX>`.   Second the `ic:`
+prefix contains an inverted index of word counts.  This was created simply to show an example of a
+follow on action when word counts change.  Ideally this follow on action would have a low chance of
+collisions.  Creating the inverted index will not cause collisions because each word is in a single
+CFM bucket and each bucket is processed independently.
+
+## Part 4 : Running this example on a real instance.
+
+Everything in the tour so far has used MiniFluo to run code.  The following
+instructions show how to run the code in this excercise on a real Fluo
+instance.  [Uno] can be used to quickly setup Fluo on a single node.
+
+The following two helper classes will be needed to run on a real instance.
+
+```java
+package ft;
+
+import org.apache.fluo.api.config.FluoConfiguration;
+
+/**
+ * Generates application config.
+ */
+public class GenConfig {
+  public static void main(String[] args) {
+    FluoConfiguration conf = new FluoConfiguration();
+    Main.preInit(conf);
+    conf.save(System.out);
+  }
+}
+```
+
+
+```java
+package ft;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.*;
+import javax.inject.Inject;
+import org.apache.fluo.api.client.*;
+import org.apache.fluo.api.config.FluoConfiguration;
+
+/**
+ * Loads one or more document passed in on the command line.
+ */
+public class Load {
+  // when run with fluo exec command, the applications configuration will be injected
+  @Inject
+  private static FluoConfiguration fluoConfig;
+
+  public static void main(String[] args) throws Exception {
+    try (FluoClient client = FluoFactory.newClient(fluoConfig);
+        LoaderExecutor loaderExecutor = client.newLoaderExecutor()) 
+    {
+      for (String filename : args) {
+        Path path = Paths.get(filename);
+        byte[] encoded = Files.readAllBytes(path);
+        String docContent = new String(encoded, StandardCharsets.UTF_8);
+        String uri = path.toAbsolutePath().normalize().toUri().toString();
+        Document doc = new Document(uri, docContent);
+        loaderExecutor.execute(new DocLoader(doc));
+      }
+    }
+  }
+}
+```
```

Comment:
Content looks okay. I haven't verified the rendering or whether the code compiles or works.